src/lib.rs

```diff
@@ -59,35 +61,35 @@ pub fn use_version(version: &String, compile: bool) -> Result<()> {
     Ok(())
 }
 
-pub fn install_version(version: &String, compile: bool) -> Result<()> {
-    println!("Installing Sui `{version}`");
+pub fn install_version(alias: &String, _compile: bool) -> Result<()> {
+    let version = fetch_version(alias)?;
+
+    println!("Installing Sui `{alias} ({version})`");
 
     #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
-    {
-        if !compile {
-            if let Ok(available_versions) = fetch_versions() {
-                if available_versions.contains(version) {
-                    return download_version(version);
-                }
-            } else {
-                println!("Could not fetch available versions");
-            }
+    if !_compile {
+        let available_versions = fetch_versions()?;
+        if available_versions.contains(&alias) {
+            download_version(&version)?;
+            println!("Downloaded Sui `{alias} ({version})`");
+            return Ok(());
         }
-
-        println!("Could not find version, assuming revision");
-        compile_version(version)?;
     }
-    #[cfg(not(all(target_os = "linux", target_arch = "x86_64")))]
-    compile_version(version)?;
+
+    compile_version(&version)?;
+    println!("Compiled Sui `{alias} ({version})`");
```

Comment:
Will say "compile" even if flag to compile is false, might be misleading/confusing